jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java

```diff
@@ -36,17 +39,19 @@
  */
 @Slf4j
 public class JaegerSpan implements Span {
+  private static final AtomicReferenceFieldUpdater<JaegerSpan, Long> durationMicrosUpdater =
```

Comment:
What is the reflection overhead, since this is on critical path?

Reply:
@yurishkuro this is a static field, so the reflection cost is paid only when the class is loaded.
As for the usage of this updater for `compareAndSet`, I saw an interesting writeup [here](https://shipilev.net/blog/2015/faster-atomic-fu/).

Reply:
That's a great article!